rgw/v2/tests/s3_swift/test_rgw_restore_index_tool.py

```diff
@@ -0,0 +1,202 @@
+                reusable.upload_mutipart_object(
+                    s3_object_name,
+                    bucket,
+                    TEST_DATA_PATH,
+                    config,
+                    user_info,
+                )
+            else:
+                if config.test_ops.get("enable_version", False):
+                    reusable.upload_version_object(
+                        config,
+                        user_info,
+                        rgw_conn,
+                        s3_object_name,
+                        config.obj_size,
+                        bucket,
+                        TEST_DATA_PATH,
+                    )
+                else:
+                    log.info("upload type: normal")
+                    reusable.upload_object(
+                        s3_object_name,
+                        bucket,
+                        TEST_DATA_PATH,
+                        config,
+                        user_info,
+                    )
+            objects_created_list.append((s3_object_name, s3_object_path))
+    if config.test_ops.get("verify_bucket_gen", False) is True:
+        time.sleep(600)
+        zone_name = config.local_zone
+        bucket_gen = reusable.fetch_bucket_gen(bucket.name)
+        log.info(f"Bucket generation value is {bucket_gen}")
+
+        bkt_stat_cmd = f"radosgw-admin bucket stats --bucket {bucket.name}"
+        num_shard = json.loads(utils.exec_shell_cmd(bkt_stat_cmd))["num_shards"]
+        num_objects = json.loads(utils.exec_shell_cmd(bkt_stat_cmd))["usage"][
+            "rgw.main"
+        ]["num_objects"]
+        bucket_id = json.loads(utils.exec_shell_cmd(bkt_stat_cmd))["id"]
+
+        log.info(
+            f"the shards and number of objects before rados clearomap are {num_shard} shards and {num_objects} objects"
+        )
+        for shard in range(num_shard):
+            cmd_rados_clear_omap = f"rados clearomap -p {zone_name}.rgw.buckets.index  .dir.{bucket_id}.{bucket_gen}.{shard}"
+            utils.exec_shell_cmd(cmd_rados_clear_omap)
+
+        bkt_stat_cmd = f"radosgw-admin bucket stats --bucket {bucket.name}"
+
+        log.info("Install the jq utility")
+        cmd = "yum install -y jq"
+        utils.exec_shell_cmd(cmd)
+        log.info(f"run the rgw-restore-bucket-index tool")
+        restore_index_cmd = f"rgw-restore-bucket-index -b {bucket.name} -y"
+        utils.exec_shell_cmd(restore_index_cmd)
+        time.sleep(60)
+        log.info(
+            "the number of objects after the restore-index tool should be same as before."
+        )
+
+        bkt_stat_cmd = f"radosgw-admin bucket stats --bucket {bucket.name}"
+        restored_num_objects = json.loads(utils.exec_shell_cmd(bkt_stat_cmd))["usage"][
+            "rgw.main"
+        ]["num_objects"]
+
+        if restored_num_objects == num_objects:
+            log.info("restore bucket index is successful!")
+        else:
+            raise AssertionError("restore bucket index is successful!")
```

Comment:
@viduship please fix this error as "restore bucket index is not successful!"

Reply:
done @hmaheswa